libcaf_net/caf/net/doorman.hpp

```diff
@@ -0,0 +1,134 @@
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include <vector>
+
+#include "caf/logger.hpp"
+#include "caf/net/make_endpoint_manager.hpp"
+#include "caf/net/socket.hpp"
+#include "caf/net/stream_socket.hpp"
+#include "caf/net/stream_transport.hpp"
+#include "caf/net/tcp_accept_socket.hpp"
+#include "caf/net/tcp_stream_socket.hpp"
+#include "caf/send.hpp"
+
+namespace caf::net {
+
+/// A doorman accepts TCP connections and creates stream_transports to handle
+/// them.
+template <class Factory>
+class doorman {
+public:
+  // -- member types -----------------------------------------------------------
+
+  using factory_type = Factory;
+
+  using application_type = typename Factory::application_type;
+
+  // -- constructors, destructors, and assignment operators --------------------
+
+  explicit doorman(net::tcp_accept_socket acceptor, factory_type factory)
+    : acceptor_(acceptor), factory_(std::move(factory)) {
+    // nop
+  }
+
+  // -- properties -------------------------------------------------------------
+
+  net::tcp_accept_socket handle() {
+    return acceptor_;
+  }
+
+  // -- member functions -------------------------------------------------------
+
+  template <class Parent>
+  error init(Parent& parent) {
+    // TODO: is initializing application factory nessecary?
+    if (auto err = factory_.init(parent))
+      return err;
+    return none;
+  }
+
+  template <class Parent>
+  bool handle_read_event(Parent& parent) {
+    auto x = net::accept(acceptor_);
+    if (!x) {
+      CAF_LOG_ERROR("accept failed:" << parent.system().render(x.error()));
+      return false;
+    }
+    // TODO this won't work anymore
```

Comment:
What is this comment all about?

Reply:
I actually don't know anymore.. I mean it works  👀